docs/design/pattern_matching.md

```diff
@@ -287,45 +294,25 @@ specified.
 
 ### `var`
 
-A `var` prefix indicates that a pattern provides mutable storage for the
-scrutinee.
+A `var` prefix indicates that binding patterns within it are variable binding
+patterns rather than value binding patterns.
 
 -   _proper-pattern_ ::= `var` _proper-pattern_
 
-A `var` pattern matches when its nested pattern matches. The type of the storage
-is the resolved type of the nested _pattern_. Any binding patterns within the
-nested pattern refer to portions of the corresponding storage rather than to the
-scrutinee.
+A `var` pattern matches when its nested pattern matches.
 
 ```carbon
 fn F(p: i32*);
 fn G() {
   match ((1, 2)) {
     // `n` is a mutable `i32`.
     case (var n: i32, 1) => { F(&n); }
-    // `n` and `m` are the elements of a mutable `(i32, i32)`.
+    // `n` and `m` are separate mutable `i32`s.
```

Comment:
I would like to see a proposal and rationale for this change. This seems like a distinct change from the change that `var` doesn't defer initialization.

Reply:
Some disadvantages of this change:
- It means that `var pattern = F();` might end up performing copies if the pattern isn't just a single binding pattern.
- If the object being matched by the `var` pattern has some reason that it needs to remain a single object, it might be surprising that it gets decomposed. (This might not be relevant until we introduce customizable pattern matching for user-defined types.)
- Treating only binding patterns as introducing variables means that other parts of the object corresponding to the `var` pattern might get destroyed earlier than intended, which might be surprising if their destruction has side-effects.
It'd be good to be able to compare those concerns to the benefits.

Reply:
OK, done.